platform/src/main/java/org/stellar/anchor/platform/custody/CustodyTransactionService.java

```diff
@@ -85,22 +92,61 @@ public CreateTransactionPaymentResponse createPayment(String txnId, String reque
       updateCustodyTransaction(txn, response.getId(), SUBMITTED);
     } catch (FireblocksException e) {
       updateCustodyTransaction(txn, StringUtils.EMPTY, CustodyTransactionStatus.FAILED);
-      switch (HttpStatus.valueOf(e.getStatusCode())) {
-        case TOO_MANY_REQUESTS:
-          throw new CustodyTooManyRequestsException(e.getRawMessage());
-        case BAD_REQUEST:
-          throw new CustodyBadRequestException(e.getRawMessage());
-        case SERVICE_UNAVAILABLE:
-          throw new CustodyServiceUnavailableException(e.getRawMessage());
-        default:
-          debugF("Unhandled status code (%s)", e.getStatusCode());
-          throw e;
-      }
+      throw (getResponseException(e));
     }
 
     return response;
   }
 
+  /**
+   * Create custody transaction refund. This method acts like a proxy. It forwards request to
+   * custody payment service, updates custody transaction and handles errors
+   *
+   * @param txnId custody/SEP transaction ID
+   * @param refundRequest {@link CreateTransactionRefundRequest} object
+   * @return external transaction payment ID
+   */
+  public CreateTransactionPaymentResponse createRefund(
+      String txnId, CreateTransactionRefundRequest refundRequest) throws AnchorException {
+    JdbcCustodyTransaction txn = custodyTransactionRepo.findBySepTxId(txnId).orElse(null);
+    if (txn == null) {
+      throw new CustodyNotFoundException(String.format("Transaction (id=%s) is not found", txnId));
+    }
+
+    JdbcCustodyTransaction refundTxn = createTransactionRefundRecord(txn, refundRequest);
+
+    CreateTransactionPaymentResponse response;
+    try {
+      response = custodyPaymentService.createTransactionRefund(refundTxn);
+      updateCustodyTransaction(refundTxn, response.getId(), SUBMITTED);
+    } catch (FireblocksException e) {
+      updateCustodyTransaction(txn, StringUtils.EMPTY, CustodyTransactionStatus.FAILED);
+      throw (getResponseException(e));
+    }
+
+    return response;
+  }
+
+  private JdbcCustodyTransaction createTransactionRefundRecord(
+      JdbcCustodyTransaction txn, CreateTransactionRefundRequest refundRequest)
+      throws CustodyBadRequestException {
+    return create(
+        CreateCustodyTransactionRequest.builder()
+            .id(txn.getSepTxId())
+            .memo(refundRequest.getMemo())
+            .memoType(refundRequest.getMemoType())
+            .protocol(txn.getProtocol())
+            .fromAccount(txn.getToAccount())
+            .toAccount(txn.getFromAccount())
+            .amount(refundRequest.getAmount())
+            .amountAsset(refundRequest.getAmountAsset())
+            .kind(
```

Comment:
Looks like, it will always be `DEPOSIT`. At least for SEP-24. This field is only used to determine `PaymentHandler`. For now, i don't know whether we will have different handlers for refunds
![image](https://github.com/stellar/java-stellar-anchor-sdk/assets/131388170/7edf46cf-08d3-43c6-9249-e2a040048610)